playbooks/common/openshift-cluster/upgrades/v3_9/label_nodes.yml

```diff
@@ -0,0 +1,30 @@
+---
+
+- name: Update node labels to differentiate from (now-schedulable) masters
+  hosts: oo_first_master
+  roles:
+    - openshift_facts
+    - lib_openshift
+  tasks:
+    - import_role:
+        name: openshift_manage_node
+        tasks_from: set_default_node_role.yml
+      vars:
+        openshift_master_host: '{{ groups.oo_first_master.0 }}'
+
+- name: Update defaultNodeSelector to target non-master nodes
+  hosts: oo_masters_to_config
+  roles:
+    - openshift_facts
+  tasks:
+    - when: osm_default_node_selector is not defined
+      block:
+        - name: Change default node selector to compute=true
+          lineinfile:
+            path: /etc/origin/master/master-config.yaml
+            line: '  defaultNodeSelector: "node-role.kubernetes.io/compute=true"'
+            regexp: '  defaultNodeSelector:.*'
+
+        - import_role:
```

Comment:
The tasks imported here seem to be skipped when I run the playbook (the block is executed though), anyone know what's up with that?

Reply:
oooh, it's skipped because openshift_master has openshift_default_node_selector defined, duh

Reply:
ah, that when conditional might be applied to the imported tasks, in and that variable is defined inside that role.

Reply:
Unforunately, we can't do this.  Users might have modified this file outside of openshift-ansible.

We need to check the value of that variable, set it if it's empty or set it if it's undefined.

Reply:
This should also be in the openshift_master role and trigger during the upgrade path.

Reply:
This is in the master role though, this is just importing the tasks from it.

Reply:
wait, what was 'this' in reference to?

Reply:
Sorry, should be the lineinfile bits.